codegen/src/test/kotlin/software/amazon/smithy/rust/codegen/smithy/protocols/InlineFunctionNamerTest.kt

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package software.amazon.smithy.rust.codegen.smithy.protocols
+
+import io.kotest.assertions.withClue
+import io.kotest.matchers.shouldBe
+import org.junit.jupiter.api.Test
+import software.amazon.smithy.rust.codegen.testutil.asSmithyModel
+import software.amazon.smithy.rust.codegen.testutil.testSymbolProvider
+import software.amazon.smithy.rust.codegen.util.lookup
+
+class InlineFunctionNamerTest {
+    private val testModel = """
+        namespace test
+
+        structure SomeStruct1 {
+            some_string: String,
+            some_int: Integer,
+        }
+
+        union SomeUnion1 {
+            int: Integer,
+            long: Long,
+        }
+
+        map SomeMap1 {
+            key: String,
+            value: SomeStruct1,
+        }
+
+        list SomeList1 {
+            member: Integer,
+        }
+
+        set SomeSet1 {
+            member: Integer,
+        }
+
+        structure Op1Input {
+            some_struct: SomeStruct1,
+            some_list: SomeList1,
+            some_set: SomeSet1,
+            some_union: SomeUnion1,
+            some_map: SomeMap1,
+        }
+
+        operation Op1 {
+            input: Op1Input,
```

Comment:
we should out an output & multiple error shapes here

Reply:
also, you set the operation `input`, but since the parser was modified, we should be testing the operation output

Reply:
I'm not sure this will actually make a difference with how this is coded up.